src/game.js

```diff
@@ -0,0 +1,112 @@
+import Tile from './tiles';
+
+class Game {
+  constructor(size) {
+    const tilesWin = [];
+    for (let i = 0; i < size * size - 1; i++) {
+      tilesWin[i] = i + 1;
+    }
+    tilesWin[size * size - 1] = '';
+
+    this.tilesWin = tilesWin;
+    this.tilesWin1514 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 15, 14, ''];
+    this.length = size * size;
+
+    this.container = document.createElement('DIV');
+    this.renderTiles = this.renderTiles.bind(this);
+    this.beginGame = this.beginGame.bind(this);
+    this.shuffle = this.shuffle.bind(this);
+    this.moveToEmpty = this.moveToEmpty.bind(this);
+    this.checkTheEndOfGame = this.checkTheEndOfGame.bind(this);
+  }
+
+  init() {
+    const gameContainer = document.querySelector('.game-begin');
+    const startGame = document.createElement('BUTTON');
+    startGame.innerHTML = 'Start new Game';
+    startGame.classList.add('start-game');
+
+    this.container.classList.add('container');
+    gameContainer.append(startGame, this.container);
+
+    if (JSON.parse(localStorage.getItem('tiles'))) {
+      this.tiles = JSON.parse(localStorage.getItem('tiles'));
+      this.renderTiles(this.tiles);
+    } else {
+      this.renderTiles(this.tilesWin);
+    }
+
+    startGame.addEventListener('click', this.beginGame);
+  }
+
+  beginGame() {
+    this.tiles = [...this.tilesWin];
+    this.shuffle();
+    this.renderTiles(this.tiles);
+  }
+
+  shuffle() {
+    for (let i = this.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      const temp = this.tiles[j];
+      this.tiles[j] = this.tiles[i];
+      this.tiles[i] = temp;
+    }
+    return this.tiles;
+  }
+
+  renderTiles(arr) {
+    this.container.innerHTML = '';
+    arr.forEach((tile) => {
+      const tileCard = new Tile(tile);
+      tileCard.render(this.container);
+    });
+
+    this.container.addEventListener('click', this.moveToEmpty);
+    localStorage.setItem('tiles', JSON.stringify(this.tiles));
+  }
+
+  async moveToEmpty(e) {
+    const moveKey = e.target.closest('DIV');
```

Comment:
From the name `moveKey` it seems that here should be key already, but the variable has reference to e.target parent
Put here either key or change name to correspond the reality :)